### DISCUSSION TITLE
How to add external blockchain in CosmosSDK?

### DISCUSSION THREAD

**Discussion by Asius22** [cosmos/community#65] [Chain Builders (Cosmos SDK)] [answered]
Hi everyone, i'm trying to develop interoperability between polygon, hyperledger aries and algorand blockchains. As i see actually the ibc protocoll is not developed for these blockchain. There are some tricks that i can use to connect these blockchains with cosmos sdk?

**Comment by tac0turtle** [comment, marked as answer]
In order to connect these chains you can launch a cosmos sdk based chain and have a committee based bridge (validator multisig) which communicates to these chains. The other option is to look at what environments they have, EVM, WASM or others and see what ibc implementations exist to use in those VMs

**Reply by Asius22** [reply]
The first option looks like to be easier. Btw algorand work with client and i cant publish smart contract. Client can be write with go and there is an official ibc library wrote in go. Polygon works with evm, there is a library wrote by hyperledger lab that implements ibc in solidity. But I dont know how to proceed

**Comment by Aeroxander** [comment]
Union.build seems to be the most promising way to connect IBC to EVM chains, pretty sure it's not difficult for them to add support for non-EVM chains
